### ISSUE TITLE
msg queue overflow when sending many messages to Tendermint

### ISSUE THREAD

**Issue by mamsen** [tendermint/tendermint#445]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source): 0.9.0

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
Own app

**Merkleeyes version** (use `git rev-parse --verify HEAD`, skip if you don't use it):

**Environment**:
- **OS** (e.g. from /etc/os-release): Linux Mint 18.1 Serena, Kernel: 4.4.0-53-generic x86_64 (64 bit), Desktop: Cinnamon 3.2.7
- **Install tools**:
- **Others**:

**What happened**:
When sending many tx to Tendermint that are a bit bigger (approximately 11 KB per message) Tendermint crashes at some point.

The message that has been sending in a loop is attached to this ticket:
[message.txt](https://github.com/tendermint/tendermint/files/913901/message.txt)

Here is the full log of Tendermint for a test scenario with 100.000 messages sent in approximately 8:25 minutes, i.e. approximately 200 messages per second.

```
NOTE[04-11|12:37:44] Loaded PrivValidator                     module=types file=/home/manuel/.tendermint/priv_validator.json privValidator="PrivValidator{E735BA2A3BF1482AB9E9BF83073BAB09163A6F43 LH:0, LR:0, LS:0}"
WARN[04-11|12:37:44] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:37:47] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:37:50] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:37:53] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:37:56] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:37:59] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
WARN[04-11|12:38:02] abci.socketClient failed to connect to tcp://192.168.56.101:46658.  Retrying... module=abcicli
NOTE[04-11|12:38:05] ABCI Handshake                           module=consensus appHeight=0 appHash=
NOTE[04-11|12:38:05] ABCI Replay Blocks                       module=consensus appHeight=0 storeHeight=0 stateHeight=0
NOTE[04-11|12:38:05] Completed ABCI Handshake - Tendermint and App are synced module=consensus appHeight=0 appHash=
NOTE[04-11|12:38:08] ConsensusReactor                         module=consensus fastSync=false
NOTE[04-11|12:38:08] Catchup by replaying consensus messages  module=consensus height=1
NOTE[04-11|12:38:08] Replay: Done                             module=consensus
NOTE[04-11|12:38:08] Starting RPC HTTP server on tcp socket 0.0.0.0:46657 module=rpcserver
NOTE[04-11|12:38:08] Started node                             module=main nodeInfo="&{PubKey:PubKeyEd25519{FAB17E9582DE6EE01284F1C51E817E371C650E9B4EB3C25903F7BAB65A6F1393} Moniker:anonymous Network:test-chain-LVgjTs RemoteAddr: ListenAddr:192.168.56.102:46656 Version:0.9.0 Other:[wire_version=0.6.0 p2p_version=0.4.0 consensus_version=v1/0.2.2 rpc_version=0.6.0/3 rpc_addr=tcp://0.0.0.0:46657]}"
NOTE[04-11|12:38:08] enterNewRound(1/0). Current: 1/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:08] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=0653A928152D001052A66AD214463667E025167F
NOTE[04-11|12:38:08] Finalizing commit of block with 0 txs    module=consensus height=1 hash=0653A928152D001052A66AD214463667E025167F root=
NOTE[04-11|12:38:08] New websocket connection                 module=rpcserver remote=192.168.56.101:52069
NOTE[04-11|12:38:09] enterNewRound(2/0). Current: 2/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:09] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=7E396A46FE0C61B704A20B935E8A2D2DF3FC5BA9
NOTE[04-11|12:38:09] Finalizing commit of block with 0 txs    module=consensus height=2 hash=7E396A46FE0C61B704A20B935E8A2D2DF3FC5BA9 root=
NOTE[04-11|12:38:10] enterNewRound(3/0). Current: 3/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:10] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=CFFD6EFCE9FA60031CBB989F93A277E7DC5F03C6
NOTE[04-11|12:38:10] Finalizing commit of block with 0 txs    module=consensus height=3 hash=CFFD6EFCE9FA60031CBB989F93A277E7DC5F03C6 root=
NOTE[04-11|12:38:11] enterNewRound(4/0). Current: 4/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:11] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=C32CC7F1207AAFA409BEA57BDD72109631182265
NOTE[04-11|12:38:11] Finalizing commit of block with 0 txs    module=consensus height=4 hash=C32CC7F1207AAFA409BEA57BDD72109631182265 root=
NOTE[04-11|12:38:12] enterNewRound(5/0). Current: 5/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:12] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A09A12C67E9FAA30B10B964CAFEBF117F9BB1965
NOTE[04-11|12:38:12] Finalizing commit of block with 0 txs    module=consensus height=5 hash=A09A12C67E9FAA30B10B964CAFEBF117F9BB1965 root=
NOTE[04-11|12:38:13] enterNewRound(6/0). Current: 6/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:13] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=CE7AC48C4BB625D768EBC43F799C3F1682262771
NOTE[04-11|12:38:13] Finalizing commit of block with 0 txs    module=consensus height=6 hash=CE7AC48C4BB625D768EBC43F799C3F1682262771 root=
NOTE[04-11|12:38:14] enterNewRound(7/0). Current: 7/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:14] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=26933D9D12935438E6735372BC9A4C64285972A5
NOTE[04-11|12:38:14] Finalizing commit of block with 0 txs    module=consensus height=7 hash=26933D9D12935438E6735372BC9A4C64285972A5 root=
NOTE[04-11|12:38:15] enterNewRound(8/0). Current: 8/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:15] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=D9934BC7FEA85025F3F26E81C1C8F7ABA25B32E1
NOTE[04-11|12:38:15] Finalizing commit of block with 0 txs    module=consensus height=8 hash=D9934BC7FEA85025F3F26E81C1C8F7ABA25B32E1 root=
NOTE[04-11|12:38:16] enterNewRound(9/0). Current: 9/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:16] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=1612017FF93F7686AC27ED09D18092F65EC92AC7
NOTE[04-11|12:38:16] Finalizing commit of block with 0 txs    module=consensus height=9 hash=1612017FF93F7686AC27ED09D18092F65EC92AC7 root=
NOTE[04-11|12:38:17] enterNewRound(10/0). Current: 10/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:17] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=4047586A3FD823F38916B7FA57AB49CA2725BAEA
NOTE[04-11|12:38:17] Finalizing commit of block with 0 txs    module=consensus height=10 hash=4047586A3FD823F38916B7FA57AB49CA2725BAEA root=
NOTE[04-11|12:38:18] enterNewRound(11/0). Current: 11/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:18] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=BA94448A69702DB81D094F52A73D9EB12BD82A0E
NOTE[04-11|12:38:18] Finalizing commit of block with 0 txs    module=consensus height=11 hash=BA94448A69702DB81D094F52A73D9EB12BD82A0E root=
NOTE[04-11|12:38:19] enterNewRound(12/0). Current: 12/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:19] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=21934AA0EEB368A1427F024E6A78B972684C752F
NOTE[04-11|12:38:19] Finalizing commit of block with 0 txs    module=consensus height=12 hash=21934AA0EEB368A1427F024E6A78B972684C752F root=
NOTE[04-11|12:38:20] enterNewRound(13/0). Current: 13/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:20] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=20907A7CE3D1C4D0D02FE13EC82759FBA5ABBF8A
NOTE[04-11|12:38:20] Finalizing commit of block with 0 txs    module=consensus height=13 hash=20907A7CE3D1C4D0D02FE13EC82759FBA5ABBF8A root=
NOTE[04-11|12:38:21] enterNewRound(14/0). Current: 14/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:21] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=F2CA72D60C2B941E277006807CC5BE1692D404E5
NOTE[04-11|12:38:21] Finalizing commit of block with 0 txs    module=consensus height=14 hash=F2CA72D60C2B941E277006807CC5BE1692D404E5 root=
NOTE[04-11|12:38:22] enterNewRound(15/0). Current: 15/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:22] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=76E7AF1969DA950C949CD2A4DD7934534022E641
NOTE[04-11|12:38:22] Finalizing commit of block with 0 txs    module=consensus height=15 hash=76E7AF1969DA950C949CD2A4DD7934534022E641 root=
NOTE[04-11|12:38:23] enterNewRound(16/0). Current: 16/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:23] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=AA5CEC2E0AFE3CE40CEC546E7A1F2B8431ABEDD4
NOTE[04-11|12:38:23] Finalizing commit of block with 0 txs    module=consensus height=16 hash=AA5CEC2E0AFE3CE40CEC546E7A1F2B8431ABEDD4 root=
NOTE[04-11|12:38:24] enterNewRound(17/0). Current: 17/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:24] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=80E5E96B0262BDBE5CC952EB2AA8CC852105230D
NOTE[04-11|12:38:24] Finalizing commit of block with 0 txs    module=consensus height=17 hash=80E5E96B0262BDBE5CC952EB2AA8CC852105230D root=
NOTE[04-11|12:38:25] enterNewRound(18/0). Current: 18/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:25] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=3C185277BC6DB04E273331525F0673A4E7F81B7E
NOTE[04-11|12:38:25] Finalizing commit of block with 0 txs    module=consensus height=18 hash=3C185277BC6DB04E273331525F0673A4E7F81B7E root=
NOTE[04-11|12:38:26] enterNewRound(19/0). Current: 19/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:26] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=51E4AA9C1338404FC8DBD98D86C692BA2192B48B
NOTE[04-11|12:38:26] Finalizing commit of block with 0 txs    module=consensus height=19 hash=51E4AA9C1338404FC8DBD98D86C692BA2192B48B root=
NOTE[04-11|12:38:27] enterNewRound(20/0). Current: 20/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:27] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=37FB815075DD5E1869F89A6A922AD87F5AE27E44
NOTE[04-11|12:38:27] Finalizing commit of block with 0 txs    module=consensus height=20 hash=37FB815075DD5E1869F89A6A922AD87F5AE27E44 root=
NOTE[04-11|12:38:28] enterNewRound(21/0). Current: 21/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:28] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=B0D12DFB151096489FAEA317781EB20243B0D008
NOTE[04-11|12:38:28] Finalizing commit of block with 0 txs    module=consensus height=21 hash=B0D12DFB151096489FAEA317781EB20243B0D008 root=
NOTE[04-11|12:38:29] enterNewRound(22/0). Current: 22/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:29] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=5C068169626E6764F3B2D35A27AFC5721365EE2F
NOTE[04-11|12:38:29] Finalizing commit of block with 0 txs    module=consensus height=22 hash=5C068169626E6764F3B2D35A27AFC5721365EE2F root=
NOTE[04-11|12:38:30] enterNewRound(23/0). Current: 23/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:30] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=C9CF58ADF406326D5233D91B08E4B3CE84CC8ED2
NOTE[04-11|12:38:30] Finalizing commit of block with 0 txs    module=consensus height=23 hash=C9CF58ADF406326D5233D91B08E4B3CE84CC8ED2 root=
NOTE[04-11|12:38:31] enterNewRound(24/0). Current: 24/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:31] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=301A3CF6D32861C784BC385806807DA34FFC1FFE
NOTE[04-11|12:38:31] Finalizing commit of block with 0 txs    module=consensus height=24 hash=301A3CF6D32861C784BC385806807DA34FFC1FFE root=
NOTE[04-11|12:38:32] enterNewRound(25/0). Current: 25/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:32] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=4810CE2F957A891EA1819C7037C0488BF7A06940
NOTE[04-11|12:38:32] Finalizing commit of block with 0 txs    module=consensus height=25 hash=4810CE2F957A891EA1819C7037C0488BF7A06940 root=
NOTE[04-11|12:38:33] enterNewRound(26/0). Current: 26/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:33] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A57EB83D84EC3AE81D305392FD00B6C0877B4160
NOTE[04-11|12:38:33] Finalizing commit of block with 72 txs   module=consensus height=26 hash=A57EB83D84EC3AE81D305392FD00B6C0877B4160 root=
NOTE[04-11|12:38:34] enterNewRound(27/0). Current: 27/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:35] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=C00DAE86243A23D84B043B0B1749DD43F1EDD5A5
NOTE[04-11|12:38:35] Finalizing commit of block with 156 txs  module=consensus height=27 hash=C00DAE86243A23D84B043B0B1749DD43F1EDD5A5 root=2EEB97701CDA58DFB1FA2E7B2D01AAD26EBFD11F
NOTE[04-11|12:38:36] enterNewRound(28/0). Current: 28/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:36] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=693B0059D7938EDB331E9923A67C1CF6ED1FAF9C
NOTE[04-11|12:38:36] Finalizing commit of block with 244 txs  module=consensus height=28 hash=693B0059D7938EDB331E9923A67C1CF6ED1FAF9C root=BB809B2C316B92CC8D7C926FA38D80B5C15D98D5
NOTE[04-11|12:38:37] enterNewRound(29/0). Current: 29/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:37] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=117FB21A207171CB012660937C3C22866C37E45C
NOTE[04-11|12:38:37] Finalizing commit of block with 314 txs  module=consensus height=29 hash=117FB21A207171CB012660937C3C22866C37E45C root=0FC1245991E12DABC752B79F42AA65D40F285969
NOTE[04-11|12:38:38] enterNewRound(30/0). Current: 30/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:39] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=2F6B7CF3751DE842F17A2CF476A44F5E7DB70F8F
NOTE[04-11|12:38:39] Finalizing commit of block with 293 txs  module=consensus height=30 hash=2F6B7CF3751DE842F17A2CF476A44F5E7DB70F8F root=E3BE456B44CE58FB675DE81A8B7739C3CF6D34C3
NOTE[04-11|12:38:40] enterNewRound(31/0). Current: 31/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:40] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=90160492CCE6E8C2DC5F5F4B8B46D0A5A423C3B8
NOTE[04-11|12:38:40] Finalizing commit of block with 342 txs  module=consensus height=31 hash=90160492CCE6E8C2DC5F5F4B8B46D0A5A423C3B8 root=2B29C46BB1F900683276D479D13FD7EB6260E01F
NOTE[04-11|12:38:41] enterNewRound(32/0). Current: 32/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:41] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=EF130F222F2311752865B5FBC61F270CC8137DD8
NOTE[04-11|12:38:41] Finalizing commit of block with 319 txs  module=consensus height=32 hash=EF130F222F2311752865B5FBC61F270CC8137DD8 root=1ED220CA031AE3305AF78DDC154066CC430932CA
NOTE[04-11|12:38:42] enterNewRound(33/0). Current: 33/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:43] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=70A9B4B4C1A962CC32CFE979C7F7A7DD6287DA5B
NOTE[04-11|12:38:43] Finalizing commit of block with 231 txs  module=consensus height=33 hash=70A9B4B4C1A962CC32CFE979C7F7A7DD6287DA5B root=23AAD9635BD6550B96FA6F9BDDB194CA00E0399B
NOTE[04-11|12:38:44] enterNewRound(34/0). Current: 34/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:44] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=D4CC786F7404431540ACCD2E6002FD3285E80934
NOTE[04-11|12:38:44] Finalizing commit of block with 309 txs  module=consensus height=34 hash=D4CC786F7404431540ACCD2E6002FD3285E80934 root=EAC3CBE4D439DC0B4E1CBF778DE02CFB424E1EF0
NOTE[04-11|12:38:45] enterNewRound(35/0). Current: 35/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:45] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=F4E3ACCB09B63E0D32F870D446438332CBE0E0B6
NOTE[04-11|12:38:45] Finalizing commit of block with 323 txs  module=consensus height=35 hash=F4E3ACCB09B63E0D32F870D446438332CBE0E0B6 root=53F69135D567ACBC05B6E5A772957A15E0DA3EAC
NOTE[04-11|12:38:46] enterNewRound(36/0). Current: 36/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:47] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=AF85FFD34DB51B0426A48CA41E2822F1FD9A0A1E
NOTE[04-11|12:38:47] Finalizing commit of block with 336 txs  module=consensus height=36 hash=AF85FFD34DB51B0426A48CA41E2822F1FD9A0A1E root=875F4346AECA985954D1754F958DF1DE4E571512
NOTE[04-11|12:38:48] enterNewRound(37/0). Current: 37/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:48] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=BD13AC4D44160263F2169781CAB3D80402035ED6
NOTE[04-11|12:38:48] Finalizing commit of block with 300 txs  module=consensus height=37 hash=BD13AC4D44160263F2169781CAB3D80402035ED6 root=90BA8904119A2B072FE2E22428D610AFD5056381
NOTE[04-11|12:38:49] enterNewRound(38/0). Current: 38/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:49] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=E2CD0245B2DB039EE097AB862B24C72C98BF272B
NOTE[04-11|12:38:49] Finalizing commit of block with 312 txs  module=consensus height=38 hash=E2CD0245B2DB039EE097AB862B24C72C98BF272B root=63AEB8B90C54EF401902606DCE6B44D1E474925D
NOTE[04-11|12:38:50] enterNewRound(39/0). Current: 39/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:51] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=324E08C6C91BD790A4B1A4656B49A60213860C45
NOTE[04-11|12:38:51] Finalizing commit of block with 299 txs  module=consensus height=39 hash=324E08C6C91BD790A4B1A4656B49A60213860C45 root=844A3B4D09D8BF062FE5DB9CE359878C44B6E5FD
NOTE[04-11|12:38:52] enterNewRound(40/0). Current: 40/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:52] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=79AEF6A7243621D6FB44E8DFEFA6DDEE7AE5DDAC
NOTE[04-11|12:38:52] Finalizing commit of block with 327 txs  module=consensus height=40 hash=79AEF6A7243621D6FB44E8DFEFA6DDEE7AE5DDAC root=5DC509B29130038B30019B6AAEDD3F2BE32845A7
NOTE[04-11|12:38:53] enterNewRound(41/0). Current: 41/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:54] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=1A935376004A2860C987CBD313A0B23E84AA0887
NOTE[04-11|12:38:54] Finalizing commit of block with 296 txs  module=consensus height=41 hash=1A935376004A2860C987CBD313A0B23E84AA0887 root=A7E7B22E84674FE4B575B7FBE1D62CBC20C0D3FF
NOTE[04-11|12:38:55] enterNewRound(42/0). Current: 42/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:55] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=CCC3C12ECE993745C4C43F5FA71F0E1A429AD99F
NOTE[04-11|12:38:55] Finalizing commit of block with 325 txs  module=consensus height=42 hash=CCC3C12ECE993745C4C43F5FA71F0E1A429AD99F root=FCE11710FD5DA8941ABB86BA5D7C765A72A9E76F
NOTE[04-11|12:38:56] enterNewRound(43/0). Current: 43/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:56] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=7A805393EDE1B7519A86CBA49D8B07D40A66BF96
NOTE[04-11|12:38:56] Finalizing commit of block with 310 txs  module=consensus height=43 hash=7A805393EDE1B7519A86CBA49D8B07D40A66BF96 root=0316745C72727B804032D24A24E70DEEE6C560C7
NOTE[04-11|12:38:57] enterNewRound(44/0). Current: 44/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:58] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=6F06FE3CF97A2103FC989FDB4A454D420CECFC23
NOTE[04-11|12:38:58] Finalizing commit of block with 363 txs  module=consensus height=44 hash=6F06FE3CF97A2103FC989FDB4A454D420CECFC23 root=5E2B7F58E4DFDA47F554B31A3A5C50F4660294BA
NOTE[04-11|12:38:59] enterNewRound(45/0). Current: 45/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:38:59] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=DD793E8F08B3AB0F30D3D7094938F79D20F258AB
NOTE[04-11|12:38:59] Finalizing commit of block with 301 txs  module=consensus height=45 hash=DD793E8F08B3AB0F30D3D7094938F79D20F258AB root=C8BB66C8F798A144D349F907CA2F37456C1383AA
NOTE[04-11|12:39:00] enterNewRound(46/0). Current: 46/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:00] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A705780645A666A528F0D56FA137BD7E80A00001
NOTE[04-11|12:39:00] Finalizing commit of block with 318 txs  module=consensus height=46 hash=A705780645A666A528F0D56FA137BD7E80A00001 root=CCDE00992CA41966D5985F8EAA67F5B1333C9678
NOTE[04-11|12:39:01] enterNewRound(47/0). Current: 47/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:02] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=724983108F73722B152AB28604B5643ED7767F3D
NOTE[04-11|12:39:02] Finalizing commit of block with 323 txs  module=consensus height=47 hash=724983108F73722B152AB28604B5643ED7767F3D root=3D792300FAD697E8D4136F38435062A335D51E31
NOTE[04-11|12:39:03] enterNewRound(48/0). Current: 48/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:03] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=95A3E96F55F7D539F4E72BFD36E5B58BFD1E7E5B
NOTE[04-11|12:39:03] Finalizing commit of block with 316 txs  module=consensus height=48 hash=95A3E96F55F7D539F4E72BFD36E5B58BFD1E7E5B root=2B444D97E16F5CBEAD4D01AC1C4C4006763278A9
NOTE[04-11|12:39:04] enterNewRound(49/0). Current: 49/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:04] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=5B78E2A6B991B8602E7E688152FE68E9E06440AE
NOTE[04-11|12:39:04] Finalizing commit of block with 278 txs  module=consensus height=49 hash=5B78E2A6B991B8602E7E688152FE68E9E06440AE root=0787A60528BB7EC79683199770AD26DEAEA4BCA2
NOTE[04-11|12:39:05] enterNewRound(50/0). Current: 50/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:06] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=BC9A0CDB416EAB845BBDB7AC66BC13678807DF7F
NOTE[04-11|12:39:06] Finalizing commit of block with 297 txs  module=consensus height=50 hash=BC9A0CDB416EAB845BBDB7AC66BC13678807DF7F root=9727A9F851FB7C566F2A9A54AFD0CF55BDE1BABE
NOTE[04-11|12:39:10] enterNewRound(51/0). Current: 51/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:12] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=AF49ADF3494A04E551AFE317BBA628E76F1059F2
NOTE[04-11|12:39:12] Finalizing commit of block with 1412 txs module=consensus height=51 hash=AF49ADF3494A04E551AFE317BBA628E76F1059F2 root=0397718FD684D1A7DFE24311988E4B16EDA177E2
NOTE[04-11|12:39:14] enterNewRound(52/0). Current: 52/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:15] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=603C23C1A41A242E45890BCA16AD231407405AEE
NOTE[04-11|12:39:15] Finalizing commit of block with 677 txs  module=consensus height=52 hash=603C23C1A41A242E45890BCA16AD231407405AEE root=84D0855F731791780899A1CA1BF42B62E563DFB7
NOTE[04-11|12:39:17] enterNewRound(53/0). Current: 53/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:17] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A820EE9850E2CD78EE24F8C4A8024688EFED5DC0
NOTE[04-11|12:39:17] Finalizing commit of block with 375 txs  module=consensus height=53 hash=A820EE9850E2CD78EE24F8C4A8024688EFED5DC0 root=51ED1D7F4E5099699B760B6A7E9FF9696641D198
NOTE[04-11|12:39:18] enterNewRound(54/0). Current: 54/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:19] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=6CAC4AAEB7E4545E691B796E4B364A8B42FD8C94
NOTE[04-11|12:39:19] Finalizing commit of block with 274 txs  module=consensus height=54 hash=6CAC4AAEB7E4545E691B796E4B364A8B42FD8C94 root=1E236601A4E0853546E45084C8C2D92E5D53CC91
NOTE[04-11|12:39:20] enterNewRound(55/0). Current: 55/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:20] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=0A0275AC6540B597036C99C4B97956991B64632F
NOTE[04-11|12:39:20] Finalizing commit of block with 325 txs  module=consensus height=55 hash=0A0275AC6540B597036C99C4B97956991B64632F root=BC0CE0F1A8BE4EB75089D0AF4056ADF1D7515316
NOTE[04-11|12:39:21] enterNewRound(56/0). Current: 56/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:21] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=208EB08D728578C568D22BC38A8218D73DB8AE8E
NOTE[04-11|12:39:21] Finalizing commit of block with 332 txs  module=consensus height=56 hash=208EB08D728578C568D22BC38A8218D73DB8AE8E root=1E9FEA4F31E90DFD017139268B359259564E9782
NOTE[04-11|12:39:23] enterNewRound(57/0). Current: 57/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:23] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=F74AF329F991E298B701E0D02709B187520CB1E9
NOTE[04-11|12:39:23] Finalizing commit of block with 396 txs  module=consensus height=57 hash=F74AF329F991E298B701E0D02709B187520CB1E9 root=0080FC6B017C667D7DEF5E58D83A464339D6F81E
NOTE[04-11|12:39:24] enterNewRound(58/0). Current: 58/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:24] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=76937D05040EEFAFB9D0E05F47FFB20BE029550A
NOTE[04-11|12:39:24] Finalizing commit of block with 258 txs  module=consensus height=58 hash=76937D05040EEFAFB9D0E05F47FFB20BE029550A root=1D09044733C9F1A50C0BEAA3B94CD112829A82D4
NOTE[04-11|12:39:25] enterNewRound(59/0). Current: 59/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:26] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=846172192AC6E49B20C33D16F2A09A75985E1320
NOTE[04-11|12:39:26] Finalizing commit of block with 348 txs  module=consensus height=59 hash=846172192AC6E49B20C33D16F2A09A75985E1320 root=B5059FF8A825AA36E46693076235594398B696E4
NOTE[04-11|12:39:27] enterNewRound(60/0). Current: 60/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:27] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=8F0BC4BCB084EFD9BDB22524B4A345B6FFBE7D8D
NOTE[04-11|12:39:27] Finalizing commit of block with 284 txs  module=consensus height=60 hash=8F0BC4BCB084EFD9BDB22524B4A345B6FFBE7D8D root=35ED10EC528A77F1EF3A3AC17C0A6C41F3731D0E
NOTE[04-11|12:39:28] enterNewRound(61/0). Current: 61/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:29] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=96EF37354BA9643182CD2148034A75EF89E22D3F
NOTE[04-11|12:39:29] Finalizing commit of block with 293 txs  module=consensus height=61 hash=96EF37354BA9643182CD2148034A75EF89E22D3F root=F42EFBBE5CBB0F214F5F6DF1B0B8143A7D811729
NOTE[04-11|12:39:30] enterNewRound(62/0). Current: 62/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:30] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=08BF9A88D55A044C664EBE405383336E2F9193FE
NOTE[04-11|12:39:30] Finalizing commit of block with 384 txs  module=consensus height=62 hash=08BF9A88D55A044C664EBE405383336E2F9193FE root=D9B8CEBD79092034B36A8E21A3461DA4E100F14B
NOTE[04-11|12:39:31] enterNewRound(63/0). Current: 63/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:31] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=F649C86DAC6834997EB9637EB8317513ABFC1D90
NOTE[04-11|12:39:31] Finalizing commit of block with 312 txs  module=consensus height=63 hash=F649C86DAC6834997EB9637EB8317513ABFC1D90 root=9D1C2D016F65450CD3B23DC812439C28F5E0153B
NOTE[04-11|12:39:32] enterNewRound(64/0). Current: 64/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:33] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=83D7A50FD3CFE3AAA959265E9F387C008045C80F
NOTE[04-11|12:39:33] Finalizing commit of block with 314 txs  module=consensus height=64 hash=83D7A50FD3CFE3AAA959265E9F387C008045C80F root=E82FFFB8CE4E79E1E082B910A73FF1E21ECFA5D5
NOTE[04-11|12:39:34] enterNewRound(65/0). Current: 65/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:34] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=D78B686A4E87678EBCF8F658B89D28A848696FCE
NOTE[04-11|12:39:34] Finalizing commit of block with 320 txs  module=consensus height=65 hash=D78B686A4E87678EBCF8F658B89D28A848696FCE root=6268BE2B948C117FE530F6753053F6A1C3E10CB8
NOTE[04-11|12:39:35] enterNewRound(66/0). Current: 66/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:36] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=95819121FED09BC021E6BB75425A30E1944B4E28
NOTE[04-11|12:39:36] Finalizing commit of block with 296 txs  module=consensus height=66 hash=95819121FED09BC021E6BB75425A30E1944B4E28 root=00D119701B693330C24F5B12D1CB3F80C296821A
NOTE[04-11|12:39:40] enterNewRound(67/0). Current: 67/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:42] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=BBF563AAC95F6C386BB11603C57FE52BBE74EB36
NOTE[04-11|12:39:42] Finalizing commit of block with 1472 txs module=consensus height=67 hash=BBF563AAC95F6C386BB11603C57FE52BBE74EB36 root=10BE571F67099F2560AD72100C25BB98B735031D
NOTE[04-11|12:39:46] enterNewRound(68/0). Current: 68/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:47] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=F315E4A742A53406CCE62BC956B48DC87C81C502
NOTE[04-11|12:39:47] Finalizing commit of block with 1048 txs module=consensus height=68 hash=F315E4A742A53406CCE62BC956B48DC87C81C502 root=4FD03EAC8C9EEF251582C8801583862A8AEAE05D
NOTE[04-11|12:39:49] enterNewRound(69/0). Current: 69/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:49] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=6A0CC0A3E0D4008C454E456143AFC8440D3BD65B
NOTE[04-11|12:39:49] Finalizing commit of block with 564 txs  module=consensus height=69 hash=6A0CC0A3E0D4008C454E456143AFC8440D3BD65B root=1F09BC55175F90181BEA1C8F23C78554F6957C12
NOTE[04-11|12:39:50] enterNewRound(70/0). Current: 70/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:51] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=20D2B3CBD6977FA0C94A9D9276393DCD3B33DCA5
NOTE[04-11|12:39:51] Finalizing commit of block with 263 txs  module=consensus height=70 hash=20D2B3CBD6977FA0C94A9D9276393DCD3B33DCA5 root=6781C42E1E8442823F58E8ECA6BFBF6F78161126
NOTE[04-11|12:39:52] enterNewRound(71/0). Current: 71/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:52] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=B365DCBFD7EEB97A74F3574BECA9971B8B9ABDD1
NOTE[04-11|12:39:52] Finalizing commit of block with 261 txs  module=consensus height=71 hash=B365DCBFD7EEB97A74F3574BECA9971B8B9ABDD1 root=65A610889989749F41A6D41AB358E93DC025AC3A
NOTE[04-11|12:39:53] enterNewRound(72/0). Current: 72/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:53] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=282733D602BEE675EE9460FD479A69067B8329C0
NOTE[04-11|12:39:53] Finalizing commit of block with 301 txs  module=consensus height=72 hash=282733D602BEE675EE9460FD479A69067B8329C0 root=FC730FDAEF19BB4EB385DE1375DA921D72ED4667
NOTE[04-11|12:39:54] enterNewRound(73/0). Current: 73/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:55] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=EB145FC2B300096FA03AEEC10A79A10EB4DE1305
NOTE[04-11|12:39:55] Finalizing commit of block with 272 txs  module=consensus height=73 hash=EB145FC2B300096FA03AEEC10A79A10EB4DE1305 root=4668CB579C43CA0A6F364DF618B19005D13A726D
NOTE[04-11|12:39:56] enterNewRound(74/0). Current: 74/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:56] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A791F47F179C25992FF8E678616ECC58A20EFD6B
NOTE[04-11|12:39:56] Finalizing commit of block with 297 txs  module=consensus height=74 hash=A791F47F179C25992FF8E678616ECC58A20EFD6B root=493CB388FCA27CDD4BF8D0FB947BD643E712F5E1
NOTE[04-11|12:39:58] enterNewRound(75/0). Current: 75/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:39:59] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=724D0C0CE493E7F31B445E47355BDD8B659F1A96
NOTE[04-11|12:39:59] Finalizing commit of block with 615 txs  module=consensus height=75 hash=724D0C0CE493E7F31B445E47355BDD8B659F1A96 root=A4BED6D54F1A29EF5C96353480586177F5927B7A
NOTE[04-11|12:40:00] enterNewRound(76/0). Current: 76/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:00] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=1EB14C152AD9AC5E7FD1F32F89466AA82E432811
NOTE[04-11|12:40:00] Finalizing commit of block with 287 txs  module=consensus height=76 hash=1EB14C152AD9AC5E7FD1F32F89466AA82E432811 root=29E265F03F6A5B430496A7D2FFF7F5C737FB133D
NOTE[04-11|12:40:01] enterNewRound(77/0). Current: 77/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:02] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=7A43FF3B64BE44696821D577DBC521924D0BAFB6
NOTE[04-11|12:40:02] Finalizing commit of block with 304 txs  module=consensus height=77 hash=7A43FF3B64BE44696821D577DBC521924D0BAFB6 root=5CA65C326BD3B5DA0DEA91A4521A0C036EF6B7C1
NOTE[04-11|12:40:03] enterNewRound(78/0). Current: 78/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:03] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=463AE8D6A43FC4CA5024BDE2ED0FEDCB1FE035F4
NOTE[04-11|12:40:03] Finalizing commit of block with 272 txs  module=consensus height=78 hash=463AE8D6A43FC4CA5024BDE2ED0FEDCB1FE035F4 root=8B97745F4179FFC288613D4E3BAB7CE326D017A1
NOTE[04-11|12:40:04] enterNewRound(79/0). Current: 79/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:04] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=FEE7F67945FBFCFDC1B0BEA27E0B022E67A7F2D8
NOTE[04-11|12:40:04] Finalizing commit of block with 277 txs  module=consensus height=79 hash=FEE7F67945FBFCFDC1B0BEA27E0B022E67A7F2D8 root=C1E54D0E288B2DF72D3670E45D622592BFA2A1CB
NOTE[04-11|12:40:05] enterNewRound(80/0). Current: 80/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:06] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=5B4C7399043F4E3AC2B224182240F03111202BE5
NOTE[04-11|12:40:06] Finalizing commit of block with 310 txs  module=consensus height=80 hash=5B4C7399043F4E3AC2B224182240F03111202BE5 root=469024C51E4116F0450CDE007A52C9FFCB064284
NOTE[04-11|12:40:07] enterNewRound(81/0). Current: 81/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:08] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=4FDF8A01AF4F9829D2D1FEE988C9BB433971DB2D
NOTE[04-11|12:40:08] Finalizing commit of block with 310 txs  module=consensus height=81 hash=4FDF8A01AF4F9829D2D1FEE988C9BB433971DB2D root=B0A8A60B811BE20513F7FE9C155A779D5544BC37
NOTE[04-11|12:40:09] enterNewRound(82/0). Current: 82/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:09] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=9698062CF0EE7608256340379576A43F73ED0352
NOTE[04-11|12:40:09] Finalizing commit of block with 276 txs  module=consensus height=82 hash=9698062CF0EE7608256340379576A43F73ED0352 root=A3B096C4BFC9EF5A1F3E8DAC6A7191BD9744218B
NOTE[04-11|12:40:10] enterNewRound(83/0). Current: 83/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:10] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=A29F69EC8BD8BA23FB2877BE5E196AAF568FCBAC
NOTE[04-11|12:40:10] Finalizing commit of block with 307 txs  module=consensus height=83 hash=A29F69EC8BD8BA23FB2877BE5E196AAF568FCBAC root=F5790CAA373CE580D0FB3BD67F985909D1608926
NOTE[04-11|12:40:11] enterNewRound(84/0). Current: 84/0/RoundStepNewHeight module=consensus
NOTE[04-11|12:40:16] enterPrecommit: +2/3 prevoted proposal block. Locking module=consensus hash=B473AFD78E7C2F6D462316BFAA750F13D7B58987
NOTE[04-11|12:40:17] Finalizing commit of block with 295 txs  module=consensus height=84 hash=B473AFD78E7C2F6D462316BFAA750F13D7B58987 root=C2FB676AB0452F19C8B0DE5866B323538BAAAFE8
NOTE[04-11|12:40:19] enterNewRound(85/0). Current: 85/0/RoundStepNewHeight module=consensus
WARN[04-11|12:40:21] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 2392"
EROR[04-11|12:40:21] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:0 P:Part{#415\n  Bytes: 202020202276...\n  Proof: SimpleProof{\n    Aunts: [D058F262B319D4E4F5BDB257D79577CB4D8E8AA9 9C48EC25508BFA1B01D1216EB2A801437879792A 6C6F5BA7D5577F164BA83A164A556D71ED0A5CCB AB44BF9428810936C6B8226A217CE345FB199C73 5AA9CA031BA249C3ADB49C7979ECB6C0950FF69C C22D4CFB872F8858D538107AD08A852A78ACB51D 4AEB0239E0E2FBB4111B9C271CCB3497BB51ECD5 139E1545C88FCEAFF4693EA50607F774249A74CA]\n  }\n}]"
NOTE[04-11|12:40:21] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:21] enterNewRound(85/1). Current: 85/0/RoundStepPrecommit module=consensus
WARN[04-11|12:40:23] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 2925"
EROR[04-11|12:40:23] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:1 P:Part{#508\n  Bytes: 363738343731...\n  Proof: SimpleProof{\n    Aunts: [3E6852F380C85F432D927977EF78D60D77FCFF65 87CB5186079823B96C0FB9FA8E51FC55DF98C7B2 0DC909277F50612C07E43F592CAD4CC506F39E4A 773FC0CC474D35A5CBA2D088FD4D540CC26DB8E5 BF0B0AA2EB2EFE3B009172DBDD22D0B0AB107DF1 B388E627EE2B45AEF7A3B8DE82A63588A089F5BE 50055AD11146E773AC6976B3A22DF87EB78A25A3 68F0BF8E1175EE9C8E5FA75293B1D008E23789A0]\n  }\n}]"
NOTE[04-11|12:40:23] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:23] enterNewRound(85/2). Current: 85/1/RoundStepPrecommit module=consensus
WARN[04-11|12:40:25] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 3524"
EROR[04-11|12:40:25] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:2 P:Part{#612\n  Bytes: 2276616C7565...\n  Proof: SimpleProof{\n    Aunts: [21BDE1C15C59CE7D9C552896AA825951417332D6 2965929C9D0E5F0929AFB535BDA3ADD2431F371E 3502E74C7AE5E4B379A84DF45F96F8165C549F7E 24302CAE8481EBD19F967867C745C9FD21B7D604 9AEC3399F0672DCBEB9931BF15ABF6976DD63AE9 6429B397D736660928009CBE03F2185604635AA7 7B590AB646D027FD18BD6CF91989BCC8ED197AA2 92B06F5625A2E43B9865B8F1FC39F04EC214E2B9 0405A974FD78D0BB356AAADABFEC2B79743B1C8A]\n  }\n}]"
NOTE[04-11|12:40:25] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:25] enterNewRound(85/3). Current: 85/2/RoundStepPrecommit module=consensus
WARN[04-11|12:40:28] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 4194"
EROR[04-11|12:40:34] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:3 P:Part{#729\n  Bytes: 202020202020...\n  Proof: SimpleProof{\n    Aunts: [910E89AC11EB71527F9E3A0D00E4C2BE4C9BE5BD 5F6CDAAFCAE8A27CF757DAF89109ADF4146347E6 B5F2566D1E4302BC19FF8A20CE41C3902182BBC5 2B83915190641E76741EFD268C1A6107591263B8 2391C20C335B3179B00AA4FFAF08EB7AD47D5025 343CFD43B16023713D7A38092A470BF9BF84D61B E76C985F23776403BC65E995691686216B125370 BAE55BFA7355B76780CA776268F219D5C568BFFD 13068366FBEBE05064DB273D51B02CD88F763389]\n  }\n}]"
NOTE[04-11|12:40:34] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:35] enterNewRound(85/4). Current: 85/3/RoundStepPrecommit module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:44] enterPrevote: ProposalBlock is nil       module=consensus
WARN[04-11|12:40:47] Internal msg queue is full. Using a go-routine module=consensus
EROR[04-11|12:40:48] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:4 P:Part{#1034\n  Bytes: 7B0A20202020...\n  Proof: SimpleProof{\n    Aunts: [C6A5B8D69F19E6CFA94AB9F31C94FCEBB1CB1BC7 380CA2D088A4F53605B46B9D16538A8745B1FD7E EB69176EC962197B1C15C2853A1A41149325176C 4FA6ECFEA22F8CEFE234196F1CFC97506CFB938C C965FD3EB499DF8DC75AEAD4D27E8E2A310ED645 0E6BCA9017AAF96D890821ACAC78FA034999ADC5 9A6BFFB91E63D1F346DE6885D6296CA44ED14752 66F0E185BC62A6ED7E25D199566DB181C3A60702 6764B21417DAF74FF31942C554021B3E14AAB15F 38054F1828711DF408B6DCB53C14CDB6DEE824B0]\n  }\n}]"
NOTE[04-11|12:40:48] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:48] enterNewRound(85/5). Current: 85/4/RoundStepPrecommit module=consensus
WARN[04-11|12:40:51] Internal msg queue is full. Using a go-routine module=consensus
WARN[04-11|12:40:51] Internal msg queue is full. Using a go-routine module=consensus
```

When Tendermint run into error with:
`WARN[04-11|12:40:21] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 2392"`

the ABCI app log says in the same moment:
```
com.google.protobuf.InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
	at com.google.protobuf.InvalidProtocolBufferException.truncatedMessage(InvalidProtocolBufferException.java:82)
	at com.google.protobuf.CodedInputStream$StreamDecoder.refillBuffer(CodedInputStream.java:2677)
	at com.google.protobuf.CodedInputStream$StreamDecoder.readRawByte(CodedInputStream.java:2737)
	at com.github.jtendermint.jabci.socket.TSocket$SocketHandler.run(TSocket.java:86)
	at java.lang.Thread.run(Thread.java:745)
```

**What you expected to happen**:
No crash

**How to reproduce it** (as minimally and precisely as possible):
Send 100.000 or more messages in the manner described to Tendermint in a running ABCI app connected to Tendermint.

**Anything else do we need to know**:

**Comment by ebuchman** [comment]
Very nice. Thanks for this! Would you mind including the piece of the log where it actually crashed?

One issue here is that we are only enforcing the MaxBlockSize (currently hard coded at 21MB) after we receive the proposal, rather than ensuring the proposal itself is within bounds. That alone would probably solve the problem.

There's also the issue of `Internal msg queue is full. Using a go-routine` which is currently marked as a TODO in the code. That said, with other constraints in place, like a proper block size, that is less of an issue.

Still curious though what the actual crash was - just OOM?

**Comment by mamsen** [comment]
Hey Ethan,
Tendermint didn't crash at all. I only snipped the following thousands of lines saying `Internal msg queue is full. Using a go-routine module=consensus` broken only sometimes with the lines
```
WARN[04-11|12:40:44] enterPrevote: ProposalBlock is nil       module=consensus
WARN[04-11|12:40:47] Internal msg queue is full. Using a go-routine module=consensus
EROR[04-11|12:40:48] Error with msg                           module=consensus type=*consensus.BlockPartMessage peer= error="Error: binary read overflow" msg="[BlockPart H:85 R:4 P:Part{#1034\n  Bytes: 7B0A20202020...\n  Proof: SimpleProof{\n    Aunts: [C6A5B8D69F19E6CFA94AB9F31C94FCEBB1CB1BC7 380CA2D088A4F53605B46B9D16538A8745B1FD7E EB69176EC962197B1C15C2853A1A41149325176C 4FA6ECFEA22F8CEFE234196F1CFC97506CFB938C C965FD3EB499DF8DC75AEAD4D27E8E2A310ED645 0E6BCA9017AAF96D890821ACAC78FA034999ADC5 9A6BFFB91E63D1F346DE6885D6296CA44ED14752 66F0E185BC62A6ED7E25D199566DB181C3A60702 6764B21417DAF74FF31942C554021B3E14AAB15F 38054F1828711DF408B6DCB53C14CDB6DEE824B0]\n  }\n}]"
NOTE[04-11|12:40:48] enterPrecommit: +2/3 prevoted for nil.   module=consensus
NOTE[04-11|12:40:48] enterNewRound(85/5). Current: 85/4/RoundStepPrecommit module=consensus
```
This was repeating endless, but Tendermint was still running. Sorry I should have mentioned that.
I can reproduce that and attach the full logfile to this ticket if this is helpfull.

In this scenario the average number of messages in one block is around 300. This makes a total size of approximately 3.300 KB plus the surrounding Tendermint stuff in the tx. To reach 21 MB almost 2.000 tx must have been accumulated for this block.
The question is: why so many tx are provided for one block when there were only in average 300 in all the preceding blocks?

In this run there are 3 outliers with significant more tx in one block:
```
Finalizing commit of block with 1412 txs module=consensus height=51
Finalizing commit of block with 1472 txs module=consensus height=67
Finalizing commit of block with 1048 txs module=consensus height=68
```

Why does this happen?

**Comment by ebuchman** [comment]
Well, glad we didn't crash. You're just running on one node right?

Indeed, most block are under the limit. But the first block failure seems to be over the limit: 

```
WARN[04-11|12:40:21] enterPrevote: ProposalBlock is invalid   module=consensus error="Wrong Block.Header.NumTxs. Expected 0, got 2392"
```

2392 txs would be over 21MB if they are all 11kB.

It's unclear why these large block outliers are showing up, but it could depend on many things. What kind of processing is your application doing? Is there any change in the rate of broadcasting transactions or is it steady ? Is it possible to see the app or its proprietary?

You can cap the number of transactions in a block with the `block_size` parameter in the config, eg. `block_size = 500` would probably solve this for you. But we'll have to dig a bit deeper to address the other issues.

**Comment by mamsen** [comment]
Hey Ethan,

I am running just one node, correct.

Does the message `Wrong Block.Header.NumTxs. Expected 0, got 2392` mean that 2392 tx are proposed for the new bock? In this case it would blast the 21 MB.

The application does no specific processing that needs lot of time. The messages come to the app, are parsed and then broadcasted to Tendermint. The rate of broadcasting is steady (happens simply in a loop) and the messages are all the same (but unique through an offset of 1 second ascending in the timestamp field).

This simple code is executed in my test scenario (Java):
```java
private void sendTestMessages() {
        Assignment assignment = new Assignment(null, null);
        assignment.setTimestamp(LocalDateTime.of(2016, 9, 15, 10, 34, 50));
        assignment.setPod("DE00106885302S1600000000056780408");
        assignment.setDate(LocalDate.of(2016, 9, 15));
        assignment.setFrom(LocalTime.of(0, 0, 0));
        assignment.setTo(LocalTime.of(0, 15, 0));

        Usage usage = new Usage();
        usage.setPod("DE00106885302S1600000000056784711");
        usage.setValue(80);
        usage.setDimension("kw");

        List<Usage> assignments = Collections.nCopies(96, usage);
        assignment.setAssignments(assignments);

        int rounds = 100000;

        for (int i = 0; i < rounds; i++) {
            String request = jsonMappingService.object2JsonString(assignment);
            assignment.setTimestamp(assignment.getTimestamp().plusSeconds(1));

            // sending request to Tendermint here
        }
    }
```

I can't give you the app, but for reproduction this is not neccessary I think.
If you take the test JSON message attached to this ticket (or use any text message of about 11 KB size) and broadcast this message 100.000 or more times to Tendermint in a loop you should be able to reproduce this behaviour very simple.

**Comment by ebuchman** [comment]
> Does the message Wrong Block.Header.NumTxs. Expected 0, got 2392 mean that 2392 tx are proposed for the new bock?

Looks like it.

> If you take the test JSON message attached to this ticket (or use any text message of about 11 KB size) and broadcast this message 100.000 or more times to Tendermint in a loop you should be able to reproduce this behaviour very simple.

Ok I'll look into replicating it today, thanks.

If you're blocked by this, you should be able to solve by just setting `block_size = 500` in the config . Then only 500 txs can be included in a proposed block. We'll look into something more robust for larger block sizes.

**Comment by ebuchman** [comment]
Great. Closing this in favor of more specific item https://github.com/tendermint/tendermint/issues/470
